middleware/authMiddleware.ts

```diff
@@ -19,24 +20,31 @@ export interface AuthMiddlewareOptions {
  * Authentication middleware that validates either:
  * 1. Valid origin + auth token, OR
  * 2. Valid API key
- * 
+ *
  * Returns the artist address and authentication method used.
  */
 export async function authMiddleware(
   req: NextRequest,
   options: AuthMiddlewareOptions = {}
 ): Promise<NextResponse | AuthResult> {
   const corsHeaders = options.corsHeaders || getCorsHeader();
-  
+
   const origin = req.headers.get("origin");
+  const referer = req.headers.get("referer");
   const authHeader = req.headers.get("authorization");
   const authToken = getBearerToken(authHeader);
   const apiKey = req.headers.get("x-api-key");
 
+  // Check origin or referer (for same-origin requests where origin might be null on mobile)
+  // Mobile browsers may not send origin header for same-origin requests
+  const hasValidOrigin =
+    isValidOrigin(origin, SITE_ORIGINAL_URL) ||
+    (!origin && referer && isValidOrigin(referer, SITE_ORIGINAL_URL));
+
   // Allow if (origin is valid AND authToken is present) OR apiKey is present
-  const hasValidOriginAndAuth = origin === SITE_ORIGINAL_URL && authToken;
+  const hasValidOriginAndAuth = hasValidOrigin && authToken;
```

Comment:
KISS principle
- why are we checking origin at all?
- You've currently chosen to add `more code` to solve the origin issue.
- MIQ: what would happen if we removed the origin check and only relied on access_token / API key for middleware?